detect/detect.go

```diff
@@ -392,6 +393,105 @@ func (d *Detector) DetectGit(source string, logOpts string, gitScanType GitScanT
 	return d.findings, nil
 }
 
+
+// DetectPipe accepts a *gitdiff.File channel which contents a git history generated from
+// the output of `git log -p ...`. startGitScan will look at each file (patch) in the history
+// and determine if the patch contains any findings.
+func (d *Detector) DetectPipe() ([]report.Finding, error) {
+
+	var (
+		err          error
+		DetectReader string
+	)
+
+	input := io.ReadCloser(os.Stdin)
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(input)
+	inputStr := buf.String()
+
+	if strings.Contains(inputStr, "diff --git") {
+		DetectReader = "git"
+	} else {
+		DetectReader = "unknown"
```

Comment:
If it's not "git" does it make sense to just treat the input as a file and scan it? Not sure how complex that is, so if you think that should be deferred for future work, that's fine.